flask/tests/conftest.py

```diff
@@ -37,6 +38,26 @@ def application():
     yield test_app
 
 
+def add_json_content_header(kwargs: dict):
+    """helper method for injecting content-type: json into headers"""
+    headers = kwargs.pop("headers", {})
+    headers["Content-Type"] = "application/json"
+    kwargs["headers"] = headers
+    return kwargs
+
+
+class TestClient(FlaskClient):
+    """ Class that adds 2 convenience methods for testing routes constrained by json content-type """
+
+    def patch_json(self, *args, **kwargs):
+        """ add content-type: json to patch request """
+        return super().patch(*args, **add_json_content_header(kwargs))
+
+    def post_json(self, *args, **kwargs):
+        """ add content-type: json to post request """
+        return super().post(*args, **add_json_content_header(kwargs))
```

Comment:
I think this might be extra as the content type is already set if the `json=` named parameter is passed to the `post` and `patch` methods.

Reply:
Gotcha. I removed the subclass and gave the tests in `test_users.py` without a `json=` explicit headers (e.g., [here](https://github.com/ccmbioinfo/stager/pull/484/commits/2dde857477321fde2f75d30373db9fa28de5cf0a#diff-ff159d3d293a3d1e6547afd4f56d2941f16ed154f21b86e61b5b5be88c48c285L158)).